### PR TITLE
Feature/control item

### PR DESCRIPTION
**Pull request by nguyenducduy1707**

### CODE REVIEW

**Comment by Winglonelion** [comment]
@nguyenducduy1707 duy  fix conflict files

File: src/reducers/todo.jsx [hunk 1]
```diff
@@ -3,7 +3,11 @@ const initialState = {
 };
 
 function nextTaskId(tasks) {
+<<<<<<< HEAD
```
Comment (Winglonelion) [inline src/reducers/todo.jsx]: what is this mean?

Please review conflict before take changed

File: src/components/InputHeader/index.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  const [text, setText] = useState('');
+
+  const dispatch = useDispatch();
+
+  const handleChange = (e) => setText(e.target.value);
+
+  const capitalizeFirstLetter = (string) => (
+    string.charAt(0).toUpperCase() + string.slice(1)
+  );
+  const handleEnterKey = (e) => {
```
Comment (Winglonelion) [inline src/components/InputHeader/index.jsx]: what is `e` ?

Comment (nguyenducduy1707) [reply src/components/InputHeader/index.jsx]: It's "event"

File: src/reducers/todo.jsx (hunk 1 above)
Comment (nguyenducduy1707) [reply src/reducers/todo.jsx]: I will fix conflict before every task on github. Thanks for your comment.

File: src/components/InputHeader/index.jsx (hunk 2 above)
Comment (Winglonelion) [reply src/components/InputHeader/index.jsx]: ok, please declare it clarify follow `Clean Code` book and push new commit

File: src/ultil.jsx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+const availableStatus = ['Not done yet', 'Done', 'In progress'];
```
Comment (Winglonelion) [inline src/ultil.jsx]: at I mentioned before, STATUS should be declared in a separate file like `config, constants` or another name of `config` file name. 

```js
export STATUSES = {
  CREATED: 'CREATED',
  IN_PROGRESS: 'IN_PROGRESS',
  DONE: 'DONE'
}
```


and show it flexible:

```js
getStatusName(status){
  switch status:
    case STATUSES.CREATED: return 'Chưa làm'
    ...
  
}

```

File: src/reducers/todo.jsx [hunk 4]
```diff
@@ -49,9 +49,15 @@ export default function todoReducer(state = initialState, action) {
         },
       };
     }
-    // case 'DELETE_TODO': {
-    //   return state.filter((todo) => todo.id !== action.payload);
-    // }
+    case 'DELETE_TODO': {
+      const { [action.payload]: taskWillBeDelete, ...tasks } = state.tasks;
```
Comment (Winglonelion) [inline src/reducers/todo.jsx]: Please make this line clear: 

What action.payload data refer to ?

if it is ID, please order your data payload like `payload: {id}` and use like
```suggestion
      const { [action.payload.id]: taskWillBeDelete, ...tasks } = state.tasks;
```

File: src/components/TodoListItem/index.jsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  const dispatch = useDispatch();
+
+  // Add description
+  const handleDescriptionChange = (e) => {
+    setDescriptionText(e.target.value);
+  };
+
+  const handleEnterAddDes = (e) => {
+    const trimedText = descriptionText.trim();
+    if (e.which === 13 && trimedText) {
```
Comment (Winglonelion) [inline src/components/TodoListItem/index.jsx]: I dont know what is this and why have this logic, Could you describe for me?

Comment (nguyenducduy1707) [reply src/components/TodoListItem/index.jsx]: That's is made for action when i press "Enter" key ( on Input "What needs to be done?" ). When i design again, i removed it. i will commit and push it as soon as possible.

File: src/reducers/todo.jsx (hunk 4 above)
Comment (nguyenducduy1707) [reply src/reducers/todo.jsx]: I get it, and i'll make it clean.

File: src/ultil.jsx (hunk 3 above)
Comment (nguyenducduy1707) [reply src/ultil.jsx]: i get it, i just push new commit. please check it when you free. It helps me a lots, thanks you so much :D

File: src/components/TodoList/index.jsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import { useSelector, shallowEqual, useDispatch } from 'react-redux';
+import { ListGroup, Button, Modal } from 'react-bootstrap';
+import TodoListItem from '../TodoListItem';
+import { addTodo } from '../../actions';
+
+const selectTodoIds = (state) => Object.values(state.todos.tasks).map((todo) => todo);
+
+function TodoList() {
+  const [show, setShow] = useState(false);
```
Comment (Winglonelion) [inline src/components/TodoList/index.jsx]:
```suggestion
  const [isShow, setIsShow] = useState(false);
```

**Pull request merged**